### PR TITLE
[backport release-1_9] Fix and improve remembered values

### PR DESCRIPTION
**Pull request by nirvn**
Manual backport.

### CODE REVIEW

File: src/core/featuremodel.h [hunk 1]
```diff
@@ -259,8 +262,9 @@ class FeatureModel : public QAbstractListModel
     GnssPositionInformation mPositionInformation;
     SnappingResult mTopSnappingResult;
     QString mTempName;
-    QMap<QgsVectorLayer *, RememberValues> mRememberings;
     bool mPositionLocked = false;
+
+    static QMutex sMutex;
```
Comment (m-kuhn) [inline src/core/featuremodel.h:267]: Actually, can we have this as Q_GLOBAL_STATIC too?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 602c933880cfdd070620812dfe8d658b8100434d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-a8b5590b4a201e8ba687443d1375b868d84d72c7-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-a8b5590b4a201e8ba687443d1375b868d84d72c7-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-a8b5590b4a201e8ba687443d1375b868d84d72c7-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-a8b5590b4a201e8ba687443d1375b868d84d72c7-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f46e3a971e5b4b6130036fe84f37d86159e7ca0b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-6acf628744ee54f6ed86442c4a0a5c06d306238a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-6acf628744ee54f6ed86442c4a0a5c06d306238a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-6acf628744ee54f6ed86442c4a0a5c06d306238a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1828-6acf628744ee54f6ed86442c4a0a5c06d306238a-x86.apk)

**Comment by nirvn** [comment]
@m-kuhn , it'll auto merge as soon as you hit approve ;)

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**